src/finance/PoolExposureSteward.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ICollector, CollectorUtils as CU} from "aave-helpers/src/CollectorUtils.sol";
+import {OwnableWithGuardian} from "solidity-utils/contracts/access-control/OwnableWithGuardian.sol";
+import {Rescuable} from "solidity-utils/contracts/utils/Rescuable.sol";
+import {RescuableBase, IRescuableBase} from "solidity-utils/contracts/utils/RescuableBase.sol";
+
+import {IPoolExposureSteward} from "./interfaces/IPoolExposureSteward.sol";
+
+/**
+ * @title PoolExposureSteward
+ * @author luigy-lemon  (Karpatkey)
+ * @author efecarranza  (Tokenlogic)
+ * @notice Manages deposits, withdrawals, and asset migrations between Aave V2 and Aave V3 assets held in the Collector.
+ *
+ * -- Security Considerations
+ *
+ * -- Pools
+ * The pools managed by the steward are allowListed.
+ * As v2 pools are deprecated, the steward only implements withdrawals from v2.
+ *
+ * -- Permissions
+ * The contract implements OwnableWithGuardian.
+ * The owner will always be the respective network Short Executor (governance).
+ * The guardian role will be given to a Financial Service provider of the DAO.
+ *
+ * While the permitted Service Provider will have full control over the funds, the allowed actions are limited by the contract itself.
+ * All token interactions start and end on the Collector, so no funds ever leave the DAO possession at any point in time.
+ */
+contract PoolExposureSteward is OwnableWithGuardian, Rescuable, IPoolExposureSteward {
+  using CU for ICollector;
+  using CU for CU.IOInput;
+
+  /// @inheritdoc IPoolExposureSteward
+  ICollector public immutable COLLECTOR;
+
+  /// @inheritdoc IPoolExposureSteward
+  mapping(address poolV3 => bool) public poolsV3;
```

Comment:
you can name the boolean as well

For the variable itself, perhaps better `approvedV3Pools` trying to be as explicit as possible?